### ISSUE TITLE
Question: EmitterSphereShape.getRandomPoint

### ISSUE THREAD

**Issue by shamanDevel** [jMonkeyEngine/jmonkeyengine#548]
This is probably a fast question:
While scanning through the code of the emitter shapes, I stumbled over the following lines of code:

```
    @Override
    public void getRandomPoint(Vector3f store) {
        do {
            store.x = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
            store.y = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
            store.z = (FastMath.nextRandomFloat() * 2f - 1f) * radius;
        } while (store.distance(center) > radius);
    }
```

https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-core/src/main/java/com/jme3/effect/shapes/EmitterSphereShape.java#L96-L103

As far as I can see it, `store` will take values withing `[-1,1]^3 * radius`. What happens if `center` is far away? Should the loop not run forever?

**Comment by Nehon** [comment]
@shamanDevel  Is this still relevant?
As a rule of thumbs, you should ask troubleshooting questions on the forum

**Comment by TehLeo** [comment]
Duplicate of #764

**Comment by shamanDevel** [comment]
And the fix is in progress, see #805

**Comment by shamanDevel** [comment]
Fixed in #805
